### DISCUSSION TITLE
Go 1.18 Generics and Endpoint

### DISCUSSION THREAD

**Discussion by shijuvar** [go-kit/kit#1221] [Q&A]
Is there any plan to implement Go 1.18 Generics in Go kit, which can reduce a lot of boilerplate of code, or may replace the Endpoint?

**Comment by sagikazarmark** [comment]
There is already a POC PR: #1189

Yes, there are plans, but they aren't clear yet.

**Comment by peterbourgon** [comment]
I've made several prototypes with generics, but none of them have delivered obvious value, or reduced boilerplate in any meaningful way.

**Reply by sagikazarmark** [reply]
I had the same experience. The only thing it improved is type safety, but I don't know if that's worth the amount of changes.

**Comment by peterbourgon** [comment]
With that said, I'd love to have more input :)
